addons/website/models/website.py

```diff
@@ -551,6 +558,140 @@ def _get_alternate_languages(self, canonical_params):
     # ----------------------------------------------------------
     # Utilities
     # ----------------------------------------------------------
+    def get_survey_logo(self):
+        if self.company_id.logo and self.company_id.logo != self.company_id._get_logo():
+            domain = [('res_model', '=', "res.partner"), ('res_field', '=', "image_1920"), ('company_id', '=', self.company_id.id)]
+            logo_attachment = self.env['ir.attachment'].sudo().search_read(domain, ['mimetype'], limit=1)
+            if logo_attachment:
+                return 'data:{};base64,{}'.format(logo_attachment[0]['mimetype'], self.company_id.logo.decode('utf-8'))
+        return False
+
+    @api.model
+    def get_survey_industries(self):
+        params = {
+            'lang': self.env.user.lang
+        }
+        resp = self.call_iap_website_service('/website/industries', params)
+        return resp['industries']
+
+    @api.model
+    def get_survey_state(self):
+        website = self.get_current_website()
+        return website.survey_state
+
+    def skip_survey(self):
+        self.survey_state = 'skipped'
+        action_id = self.env['ir.model.data'].xmlid_to_res_id('website.theme_install_kanban_action')
+        return '/web#action={}&model=ir.module.module&view_type=kanban&cids={}'.format(action_id, self.company_id.id)
+
+    def call_iap_website_service(self, route, params):
```

Comment:
Add a try/except ? What if error when calling iap ? Skip survey and go to theme selection screen ?